### PR TITLE
Add support for importing and parsing files from frameworks

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

I first want to start this by giving huge thanks to @matthewp for helping me out with this because between the Document, DocumentSnapshot, DocumentFragmentSnapshot and other Java-like classes I was (and still fairly is) lost most of the time, he really helped me out figure out how to do this!

This PR adds support for importing Svelte and Vue files and for Svelte files especially, converting them to TSX and getting proper props information, like we do for Astro files. 

This setup a bigger plan to eventually allow users to add support for any kind of framework that doesn't use JSX/TSX files without needing to make changes to the language-server itself. We're not there just yet (and we probably want to make things a lot cleaner before we do that) but, this is a first step

Additionally, this PR fix needing to remove the extension from `.ts` file in order to get types and completions. Technically, it is not wrong from TypeScript to require the extension to be removed as it is its normal behavior. However, Astro (by the past at least) as always recommended to put the extension, so we'll match that here. The way it works on the technical side is that we pretend that `.ts` files are `.tsx` files. I'm not sure why it works, but it does

I am not proud of the code but, it works and it is an improvement. Notably, the error when importing Svelte and Vue files was one of the most mentioned issues in our support threads last month

Fixes https://github.com/withastro/language-tools/issues/142 https://github.com/withastro/language-tools/issues/126

#### Diagnostics and hover information is currently unavailable for imported framework components

Not sure why, my guess is that TypeScript doesn't get the DocumentSnapshot of our Svelte file. I'm not sure why, it should technically be going through our SnapshotManager and get the proper Snapshot for said file but it seems like it doesn't. Completions however are working nicely because they go through a different system (which maybe diagnostics could also use, not sure)

#### Types from imported Vue and Svelte files are not updated when those files are changed

Again not sure why, I think this is part of a potentially bigger issue related to cache (ex https://github.com/withastro/language-tools/issues/106) or maybe it's just a question of getting the language-server to update its cache when those files are updated

## Testing

Tested manually for now. When I figure out how to make diagnostics works (in a later PR), I plan to also write tests for diagnostics for different framework's components

## Docs

Technically a bug fix so no docs needed

### CODE REVIEW

File: packages/language-server/package.json [hunk 1]
```diff
@@ -20,6 +19,7 @@
     "test": "tap --no-coverage test/**"
   },
   "dependencies": {
+    "@astrojs/svelte-language-integration": "*",
```
Comment (matthewp) [inline packages/language-server/package.json:22]: Shouldn't this be a semver range?

Comment (matthewp) [inline packages/language-server/package.json:22]: Shouldn't this be a semver range?

Comment (Princesseuh) [reply packages/language-server/package.json:22]: I wasn't really sure on how to proceed for an internal package like this, I hesitated with `latest` but I read somewhere that `latest` sometimes has issues in workspaces. We could just put the version of the package, for sure

Comment (matthewp) [reply packages/language-server/package.json:22]: I think that's what we want, look at how we do it in the main `astro` package for example. Changesets will handle updating the version when it changes.

**Pull request merged**